source/geometries/FullRingInfinity.cc

```diff
@@ -422,6 +464,117 @@ void FullRingInfinity::BuildSensors()
   }
 }
 
+void FullRingInfinity::BuildWires()
+{
+  // Add simple detector for charge
+  G4Box* chdet_solid = new G4Box("WIRE", wire_pitch_/2.,
+                                 axial_length_/2., chdet_thickn_/2);
+  G4LogicalVolume* chdet_logic =
+    new G4LogicalVolume(chdet_solid, LXe_, "WIRE");
+
+  G4String sdname = "/WIRE/ChargeDet";
+  G4SDManager* sdmgr = G4SDManager::GetSDMpointer();
+  if (!sdmgr->FindSensitiveDetector(sdname, false))
+    {
+      ChargeSD* chargesd = new ChargeSD(sdname);
+      chargesd->SetTimeBinning(wire_time_bin_);
+      G4SDManager::GetSDMpointer()->AddNewDetector(chargesd);
+      chdet_logic->SetSensitiveDetector(chargesd);
+    }
+
+  G4VisAttributes wire_col = nexus::Yellow();
+  wire_col.SetForceSolid(true);
+  chdet_logic->SetVisAttributes(wire_col);
+
+  G4double chdet_radius = inner_radius_ + lxe_depth_ - chdet_thickn_ / 2. - chdet_offset_;
+  G4int n_wires = 2. * pi * chdet_radius / wire_pitch_;
+  G4cout << "Number of wires: " << n_wires << G4endl;
+
+  G4double chdet_z_pos = 0.;
+  G4ThreeVector chdet_position(0., chdet_radius, chdet_z_pos);
+  G4int chdet_copy_no = 0;
+  G4String chdet_vol_name = "WIRE_" + std::to_string(chdet_copy_no);
+  G4RotationMatrix rot;
+  rot.rotateX(-pi / 2.);
+  new G4PVPlacement(G4Transform3D(rot, chdet_position), chdet_logic,
+                    chdet_vol_name, active_logic_, false, chdet_copy_no, false);
+
+  G4double step = 2. * pi / n_wires;
+  for (G4int i = 1; i < n_wires; ++i) {
+    G4double angle = i * step;
+    rot.rotateZ(step);
+    chdet_position.setX(-chdet_radius * sin(angle));
+    chdet_position.setY(chdet_radius * cos(angle));
+    chdet_copy_no = chdet_copy_no + 1;
+    chdet_vol_name = "WIRE_" + std::to_string(chdet_copy_no);
+    new G4PVPlacement(G4Transform3D(rot, chdet_position), chdet_logic,
+                      chdet_vol_name, active_logic_, false, chdet_copy_no, false);
+  }
+}
+
+void FullRingInfinity::BuildSeparators()
+{
+  // Separate LXe volume in smaller areas with teflon panels
+  G4Material* teflon = G4NistManager::Instance()->FindOrBuildMaterial("G4_TEFLON");
+
+  //G4double arc_sep_phi = 2 * pi * (inner_radius_ + lxe_depth_) / n_sep_phi_;
+  //G4int n_sipm_in_sep_phi = arc_sep_phi / sipm_pitch_;
+  G4double sep_angle = 2 * pi / n_sep_phi_;
+
+  G4double segm_sep_z  = axial_length_ / n_sep_z_;
+  //G4double sep_pitch_z = G4int(segm_sep_z / sipm_pitch_) * sipm_pitch_;
+
+  G4double sep_thickn = 5.*mm;
+  G4double sep_offset = 0.01*mm;
+  G4double sep_size   = lxe_depth_ - chdet_thickn_ - chdet_offset_ - sep_offset;
+  G4Tubs* sep_z_solid =
+    new G4Tubs("SEPARATOR_Z", inner_radius_, inner_radius_ + sep_size,
+               sep_thickn/2., 0, twopi);
+  G4Box* sep_phi_solid =
+    new G4Box("SEPARATOR_PHI", sep_thickn/2., sep_size/2., axial_length_/2.);
+
```

Comment:
It is not clear to me how the separations are. Do they contain only one wire or some of them?